api/assessment/metric.go

```
@@ -142,8 +170,53 @@ func (m *Metric) Validate(opts ...MetricValidationOption) (err error) {
 	return nil
 }
 
+// Validate validates the metric configuration
+func (c *MetricConfiguration) Validate() (err error) {
+
+	// isDefault does not need to be checked.
+	// UpdatedAt does not need to be checked.
+
+	if c == nil {
+		return ErrMetricConfigurationMissing
+	}
+
+	if c.Operator == "" {
+		return ErrMetricConfigurationOperatorMissing
+	}
+
+	if c.TargetValue == nil {
+		return ErrMetricConfigurationTargetValueMissing
+	}
+
+	if c.MetricId == "" {
+		return ErrMetricIdMissing
+	}
+
+	err = CheckCloudServiceID(c.CloudServiceId)
+	if err != nil {
+		return fmt.Errorf("%w", err)
```

We can just return the error here without any wrapping I think